integration/test_list.py

```diff
@@ -0,0 +1,42 @@
+
+import pytest_twisted
+
+from eliot import (
+    start_action,
+)
+
+import util
+
+# see "conftest.py" for the fixtures (e.g. "magic_folder")
+
+
+@pytest_twisted.inlineCallbacks
+def test_list(request, reactor, temp_dir, introducer_furl, flog_gatherer):
+    """
+    'magic-folder list' happy-path works
+    """
+
+    with start_action(action_type=u"integration:test_list:zelda", include_args=[], include_result=False):
+        zelda = yield util.MagicFolderEnabledNode.create(
+            reactor,
+            request,
+            temp_dir,
+            introducer_furl,
+            flog_gatherer,
+            name="zelda",
+            tahoe_web_port="tcp:9982:interface=localhost",
+            magic_folder_web_port="tcp:19982:interface=localhost",
+            storage=True,
+        )
+
+    proto = util._CollectOutputProtocol()
+    util._magic_folder_runner(
+        proto, reactor, request,
+        [
+            "--config", zelda.magic_config_directory,
+            "list",
+        ],
+    )
+    output = yield proto.done
+    assert output.strip() == "No magic-folders"
```

Comment:
I kept this in because it's what I wrote first (and it fixed some other broken stuff in the integration tests). Can we keep both kinds of tests? (There are unit-tests as well).

Reply:
I don't think this kind of test adds a lot of value compared to its cost.  It's fragile (it encodes a machine dependency on a human-facing piece of english prose) and slow to run (14 seconds for this single test on my laptop).  It may also be hard to debug (often these tells tell you just one thing "it worked" or "it didn't" and you're left to try to figure out why;  I think because of the `--debug`/traceback change this one might sometimes give you more, though I don't think that's necessarily the case - it depends where the problem occurs).  If there's a clear path to a testing strategy without these drawbacks we should take it.